src/coreclr/tools/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -4816,11 +4819,16 @@ int MethodContext::repGetStringLiteral(CORINFO_MODULE_HANDLE module, unsigned me
     {
         DD value = GetStringLiteral->Get(key);
         DEBUG_REP(dmpGetStringLiteral(key, value));
-        if (buffer != nullptr)
+        int srcBufferLength = (int)value.A;
+        if (buffer != nullptr && srcBufferLength != -1)
         {
-            memcpy(buffer, GetStringLiteral->GetBuffer(value.B), bufferSize * sizeof(char16_t));
+            char16_t* srcBuffer = (char16_t*)GetStringLiteral->GetBuffer(value.B);
+            if (srcBuffer != nullptr)
```

Comment:
assert instead? Otherwise we will be not write anything but return a length as if we did.

Reply:
Should probably throw "missing" in this case actually

Reply:
assert should be fine I guess, in case of missing context we won't have length too, right?